### PR TITLE
Improve table viz

### PR DESCRIPTION
**Pull request by Henry-Pulver**
Test with `@superflows/chat-ui-react@1.1.1-beta.1`

This does a lot more than it says on the tin. It also:
1. Adds an example nextjs project in `examples` so we can easily test
2. Adds `husky`
3. Adds `jest`
4. Fixes a minor sidebar button colour bug

Example table below:
Screenshot from 2023-08-04 17-39-07

### CODE REVIEW

File: .husky/pre-push [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npm test -- --runInBand --watchAll=false --bail --ci
```
Comment (neuromantic99) [inline .husky/pre-push:4]: we don't actually need to run tests in band this was just for when we were testing the database in the email product

File: item.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+src/something/lib
+src/something/core
+where/src/something/else
```
Comment (neuromantic99) [inline item.txt:3]: what be this?

File: examples/nextjs/package.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "@superflows/chat-ui-react": "file:../../superflows-chat-ui-react-1.1.1-48.tgz",
```
Comment (neuromantic99) [inline examples/nextjs/package.json:12]: do we still need this locally, can we push it up to npm?

File: src/lib/utils.ts [hunk 4]
```diff
@@ -1,68 +1,116 @@
...
+    // And was the holy Lamb of God,
+    return output + tablemark(functionOutput as any[], { columns });
   } else {
-    // Format: {data: {a, b}}
-    if ("data" in functionOutput) {
-      functionOutput = functionOutput.data;
-    }
     // Format: {a, b}
     Object.entries(functionOutput).forEach(([key, value]) => {
       output += `${functionNameToDisplay(key)}: ${
```
Comment (Henry-Pulver) [inline src/lib/utils.ts:116]: Oops - will fix this over the weekend!

File: package.json [hunk 5]
```diff
@@ -1,33 +1,47 @@
...
+    "husky": "^8.0.3",
+    "jest": "^29.6.2",
+    "jest-environment-jsdom": "^29.6.2",
+    "prettier": "^3.0.0",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "rollup": "^3.26.3",
+    "rollup-plugin-dts": "^5.3.1",
+    "rollup-plugin-postcss": "^4.0.2",
+    "rollup-plugin-tailwindcss": "file:../rollup-plugin-tailwindcss/rollup-plugin-tailwindcss-1.1.0.tgz",
```
Comment (neuromantic99) [inline package.json:21]: as above

File: src/components/SuperflowsSidebar.tsx [hunk 6]
```diff
@@ -363,7 +362,7 @@ export default function SuperflowsSidebar(props: {
                         ) {
                           if (
                             devChatContents[idx - 1].role === "function" ||
-                            devChatContents[idx + 1].role === "function"
+                            devChatContents[idx + 1]?.role === "function"
```
Comment (neuromantic99) [inline src/components/SuperflowsSidebar.tsx:365]: is this a fix for the serious crashing we were seeing?

Comment (Henry-Pulver) [reply src/components/SuperflowsSidebar.tsx:365]: Yep

**Pull request merged**